sdk/identity/identity/src/credentials/clientCertificateCredential.ts

```diff
@@ -28,23 +54,32 @@ export class ClientCertificateCredential implements TokenCredential {
    *
    * @param tenantId - The Azure Active Directory tenant (directory) ID.
    * @param clientId - The client (application) ID of an App Registration in the tenant.
-   * @param certificatePath - The path to a PEM-encoded public/private key certificate on the filesystem.
+   * @param configuration - Other parameters required, including the PEM-encoded certificate as a string, or as a path on the filesystem.
+   *                        If the type is ignored, we will throw if both the value of the PEM certificate and the path to a PEM certificate are provided at the same time.
    * @param options - Options for configuring the client which makes the authentication request.
    */
   constructor(
     tenantId: string,
     clientId: string,
-    certificatePath: string,
+    configuration: ClientCertificateCredentialPEMConfiguration,
     options: ClientCertificateCredentialOptions = {}
   ) {
-    if (!tenantId || !clientId || !certificatePath) {
+    if (!tenantId || !clientId) {
+      throw new Error(`${credentialName}: tenantId and clientId are required parameters.`);
+    }
+    if (!configuration || !(configuration.certificate || configuration.certificatePath)) {
+      throw new Error(
+        `${credentialName}: Provide either a PEM certificate in string form, or the path to that certificate in the filesystem.`
+      );
+    }
+    if (configuration.certificate && configuration.certificatePath) {
       throw new Error(
-        "ClientCertificateCredential: tenantId, clientId, and certificatePath are required parameters."
+        `${credentialName}: To avoid unexpected behaviors, providing both the contents of a PEM certificate and the path to a PEM certificate is forbidden. Please provide either of both.`
```

Comment:
nit: I find the "Please provide either of both." sentence at the end of this error message confusing. I think it could be removed from the message.